shared/js/background/amp-protection.es6.js

```diff
@@ -0,0 +1,183 @@
+const Site = require('./classes/site.es6')
+const tdsStorage = require('./storage/tds.es6')
+const utils = require('./utils.es6')
+
+let ampSettings = null
+
+const featureName = 'ampLinks'
+
+/**
+ * Ensure the config has the ampLinks feature and its settings
+ *
+ * @returns {boolean} - true if the config is loaded
+ */
+function ensureConfig () {
+    if (ampSettings) {
+        return true
+    }
+
+    if (!tdsStorage.config ||
+        !tdsStorage.config.features ||
+        !tdsStorage.config.features.ampLinks ||
+        !tdsStorage.config.features.ampLinks.settings ||
+        !tdsStorage.config.features.ampLinks.settings.linkFormats ||
+        !tdsStorage.config.features.ampLinks.settings.keywords) {
+        return false
+    }
+
+    ampSettings = tdsStorage.config.features.ampLinks.settings
+
+    return true
+}
+
+/**
+ * This method checks if the given url is a Google hosted AMP url and resturns the canonical url if it is.
+ *
+ * @param {Site} site - the initiating site
+ * @param {string} url - the url being loaded
+ * @returns canonical url if found, null otherwise
+ */
+function extractAMPURL (site, url) {
+    if (!ensureConfig()) {
+        return null
+    }
+
+    if (site.specialDomainName || !site.isFeatureEnabled(featureName)) {
+        return null
+    }
+
+    for (const regexPattern of ampSettings.linkFormats) {
+        const match = url.match(regexPattern)
+        if (match && match.length > 1) {
+            const newSite = new Site(match[1].startsWith('http') ? match[1] : `https://${match[1]}`)
+
+            if (newSite.specialDomainName || !newSite.isFeatureEnabled(featureName)) {
+                return null
+            }
+
+            return newSite.url
+        }
+    }
+
+    return null
+}
+
+/**
+ * Check if the given request is a suspected 1st party AMP URL. If it is, then we will attempt to fetch the canonical URL
+ *
+ * @param {object} requestData - request data
+ * @param {object} thisTab - current tab
+ * @param {object} mainFrameRequestURL - main frame request url
+ * @returns true if the request is a suspected 1st party AMP url
+ */
+function tabNeedsDeepExtraction (requestData, thisTab, mainFrameRequestURL) {
+    if (utils.getBrowserName() !== 'moz') {
```

Comment:
Can you file a follow up to investigate if there's other solutions here and remove the dead code pretty swiftly if not?

Reply:
Follow up was planned as part of postmortem process 👍